src/materialized/tests/sql.rs

```diff
@@ -345,6 +345,27 @@ fn test_tail_fetch_timeout() -> Result<(), Box<dyn Error>> {
     let rows = client.query("FETCH c WITH (TIMEOUT = '0s')", &[])?;
     assert_eq!(rows.len(), 0);
 
+    // Make a third cursor. Fetch should return immediately if there are enough
+    // rows, even with a really long timeout.
+    //
+    // Regression test for #6307
+    client.batch_execute(
+        "CLOSE c;
+        DECLARE c CURSOR FOR TAIL t",
+    )?;
+    let before = Instant::now();
+    // NB: This timeout is chosen such that the test will timeout if the bad
+    // behavior occurs.
+    let rows = client.query("FETCH 3 c WITH (TIMEOUT = '1h')", &[])?;
+    let duration = before.elapsed();
+    if rows.len() != 0 {
```

Comment:
Is it legitimate for the `FETCH` to return 0 rows in this example? the way the code is structured it seems like if we return 0 rows, the test will pass, and we'll only assert that the number of rows and their contents matches expectations if some rows are returned unless I'm deeply misreading here.

Reply:
Brain fart. Fixed